backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/AutomaticAnalysisEnablementRequestHandler.java

```diff
@@ -0,0 +1,110 @@
+import org.apache.hc.core5.http.Method;
+import org.apache.hc.core5.http.io.HttpRequestHandler;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.apache.hc.core5.net.URIBuilder;
+import org.sonarsource.sonarlint.core.analysis.AnalysisService;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+public class AutomaticAnalysisEnablementRequestHandler implements HttpRequestHandler {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private final AnalysisService analysisService;
+  private final Gson gson = new Gson();
+
+  public AutomaticAnalysisEnablementRequestHandler(AnalysisService analysisService) {
+    this.analysisService = analysisService;
+  }
+
+  @Override
+  public void handle(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context) throws HttpException, IOException {
+    if (!Method.POST.isSame(request.getMethod())) {
+      response.setCode(HttpStatus.SC_BAD_REQUEST);
+      return;
+    }
+
+    var params = new HashMap<String, String>();
+    try {
+      new URIBuilder(request.getUri(), StandardCharsets.UTF_8)
+        .getQueryParams()
+        .forEach(p -> params.put(p.getName(), p.getValue()));
+    } catch (URISyntaxException e) {
+      response.setCode(HttpStatus.SC_BAD_REQUEST);
+      var errorResponse = new AutomaticAnalysisEnablementResponse(false, "Invalid URI");
+      response.setEntity(new StringEntity(gson.toJson(errorResponse), ContentType.APPLICATION_JSON));
+      return;
+    }
+
+    var enabledParam = params.get("enabled");
+    if (enabledParam == null) {
+      response.setCode(HttpStatus.SC_BAD_REQUEST);
+      var errorResponse = new AutomaticAnalysisEnablementResponse(false, "Missing 'enabled' query parameter");
+      response.setEntity(new StringEntity(gson.toJson(errorResponse), ContentType.APPLICATION_JSON));
+      return;
+    }
+
+    boolean enabled;
+    try {
+      enabled = Boolean.parseBoolean(enabledParam);
+    } catch (Exception e) {
+      response.setCode(HttpStatus.SC_BAD_REQUEST);
+      var errorResponse = new AutomaticAnalysisEnablementResponse(false, "Invalid 'enabled' parameter value");
+      response.setEntity(new StringEntity(gson.toJson(errorResponse), ContentType.APPLICATION_JSON));
+      return;
+    }
+
+    try {
+      var result = handleAutomaticAnalysisEnablement(enabled);
+      response.setCode(HttpStatus.SC_OK);
+      response.setEntity(new StringEntity(gson.toJson(result), ContentType.APPLICATION_JSON));
+    } catch (Exception e) {
+      LOG.error("Failed to change automatic analysis", e);
+      response.setCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+      var errorResponse = new AutomaticAnalysisEnablementResponse(false, "Failed to change automatic analysis: " + e.getMessage());
+      response.setEntity(new StringEntity(gson.toJson(errorResponse), ContentType.APPLICATION_JSON));
```

Comment:
The 4 error cases look pretty similar, maybe this could be factorized in a method